src/libraries/Microsoft.PowerFx.Core/Functions/TexlFunctionSet.cs

```diff
@@ -28,45 +32,68 @@ internal class TexlFunctionSet
         // Count of functions
         internal int _count;
 
-        internal Dictionary<string, List<TexlFunction>>.KeyCollection FunctionNames => _functions.Keys;
+        internal bool _canremove;
```

Comment:
I think this is a good idea. 
A concept of immutable sets is good and can help with optimizations.